### PR TITLE
[CDAP-17928] We use hierarchical naming as the record name to ensure each record will ha…

### PR DESCRIPTION
**Pull request by sechegaray**
…ve a unique name as AVRO doesn't allow the redefinition of existing records. fixing: CDAP-17928

### CODE REVIEW

**Review by tivv — COMMENTED** [review]
I don't really like randomness here as it may bring up different schemas in various cases, e.g. different schema for rows or splits. Even if it's not like it right now it's very easy to get into this situation in future. Could you please instead use some stable names, e.g. record.field1.field2 if we allow namespaces or record_field1_field2 if we don't

Also please add a test

File: src/main/java/io/cdap/plugin/gcp/bigquery/util/BigQueryUtil.java [hunk 1]
```diff
@@ -202,7 +203,20 @@ public static Schema getTableSchema(com.google.cloud.bigquery.Schema bqSchema, @
...
+    return getSchemaField(field, collector, null);
+  }
+
+    /**
+     * Converts BigQuery schema field into a corresponding CDAP Schema.Field.
+     * @param field BigQuery field to be converted.
+     * @param collector Failure collector to collect failure messages for the client.
+     * @param recordPrefix String to prepend to recordNames to make them unique
+     * @return A CDAP schema field
+     */
```
Comment (tivv) [inline src/main/java/io/cdap/plugin/gcp/bigquery/util/BigQueryUtil.java]: It looks like indentation is wrong

File: src/main/java/io/cdap/plugin/gcp/bigquery/util/BigQueryUtil.java [hunk 2]
```diff
@@ -202,7 +203,20 @@ public static Schema getTableSchema(com.google.cloud.bigquery.Schema bqSchema, @
...
+  @Nullable
+  public static Schema.Field getSchemaField(Field field, @Nullable FailureCollector collector,
+      @Nullable String recordPrefix) {
```
Comment (tivv) [inline src/main/java/io/cdap/plugin/gcp/bigquery/util/BigQueryUtil.java]: Please doublecheck you included intellij settings from https://cdap.atlassian.net/wiki/spaces/CE/pages/1594098619/Coding+Standard and reformat according to the standard

Comment (sechegaray) [reply src/main/java/io/cdap/plugin/gcp/bigquery/util/BigQueryUtil.java]: Weird that it did not raise an issue with this.
I've reloaded IntelliJ code format settings and run "reformat code" in the file.

File: src/main/java/io/cdap/plugin/gcp/bigquery/util/BigQueryUtil.java [hunk 3]
```diff
@@ -194,15 +195,31 @@ public static Schema getTableSchema(com.google.cloud.bigquery.Schema bqSchema, @
     return Schema.recordOf("output", schemafields);
   }
 
+
```
Comment (tivv) [inline src/main/java/io/cdap/plugin/gcp/bigquery/util/BigQueryUtil.java]: nit: extra line break

File: src/main/java/io/cdap/plugin/gcp/bigquery/util/BigQueryUtil.java [hunk 4]
```diff
@@ -228,14 +245,31 @@ public static Schema getTableSchema(com.google.cloud.bigquery.Schema bqSchema, @
     return null;
   }
 
+
```
Comment (tivv) [inline src/main/java/io/cdap/plugin/gcp/bigquery/util/BigQueryUtil.java]: Also extra line break

File: src/main/java/io/cdap/plugin/gcp/bigquery/util/BigQueryUtil.java [hunk 5]
```diff
@@ -228,14 +245,31 @@ public static Schema getTableSchema(com.google.cloud.bigquery.Schema bqSchema, @
...
+   * @param field     Bigquery field to be converted.
    * @param collector Failure collector to collect failure messages for the client.
    * @return A CDAP field schema
    */
   @Nullable
   public static Schema convertFieldType(Field field, @Nullable FailureCollector collector) {
+    return convertFieldType(field, collector, null);
+  }
+
+
```
Comment (tivv) [inline src/main/java/io/cdap/plugin/gcp/bigquery/util/BigQueryUtil.java]: One more. Is reformat doing it?

File: src/test/java/io/cdap/plugin/gcp/bigquery/util/BigQueryUtilTest.java [hunk 6]
```diff
@@ -80,7 +84,18 @@ public void testGetTableSchema() {
...
+    Schema.Field cdapRecordField = Schema.Field.of("structure",
+        Schema.nullableOf(Schema.recordOf("structure", cdapSubRecordField)));
+    cdapFieldList.add(cdapRecordField);
+
     Schema expectedSchema = Schema.recordOf("output", cdapFieldList);
     assertEquals(expectedSchema, tableSchema);
   }
+
+
+
```
Comment (tivv) [inline src/test/java/io/cdap/plugin/gcp/bigquery/util/BigQueryUtilTest.java]: three empty lines here

**Review by tivv — APPROVED** [review]

**Pull request merged**